src/Altinn.Platform/Altinn.Platform.Authorization/UnitTest/Utils/AssertionUtil.cs

```diff
@@ -64,5 +86,140 @@ private static void AssertEqual(XacmlAttributeAssignment expected, XacmlAttribut
+        }
+
+        private static void AssertAttributeValueEqual(XacmlAttributeValue expected, XacmlAttributeValue actual)
+        {
+            Assert.NotNull(actual);
+            Assert.NotNull(expected);
+
+            Assert.Equal(expected.DataType.OriginalString, actual.DataType.OriginalString);
+            Assert.Equal(expected.Value, actual.Value);
+            AssertCollections(expected.Attributes, actual.Attributes, AssertXAttributeEqual);
+            AssertCollections(expected.Elements, actual.Elements, AssertXElementEqual);
+        }
+
+        private static void AssertAttributeDesignatorEqual(XacmlAttributeDesignator expected, XacmlAttributeDesignator actual)
+        {
+            Assert.NotNull(actual);
+            Assert.NotNull(expected);
+
+            Assert.Equal(expected.AttributeId.OriginalString, actual.AttributeId.OriginalString);
+            Assert.Equal(expected.Category.OriginalString, actual.Category.OriginalString);
+            Assert.Equal(expected.DataType.OriginalString, actual.DataType.OriginalString);
+            Assert.Equal(expected.Issuer, actual.Issuer);
+            Assert.Equal(expected.MustBePresent, actual.MustBePresent);
+        }
+
+        private static void AssertObligationExpressionEqual(XacmlObligationExpression expected, XacmlObligationExpression actual)
+        {
+            Assert.NotNull(actual);
+            Assert.NotNull(expected);
+
+            Assert.Equal(expected.ObligationId.OriginalString, actual.ObligationId.OriginalString);
+            Assert.Equal(expected.FulfillOn, actual.FulfillOn);
+
+            AssertCollections(expected.AttributeAssignmentExpressions, actual.AttributeAssignmentExpressions, AssertAttributeAssignmentExpressionEqual);
+        }
+
+        private static void AssertAttributeAssignmentExpressionEqual(XacmlAttributeAssignmentExpression expected, XacmlAttributeAssignmentExpression actual)
+        {
+            Assert.NotNull(actual);
+            Assert.NotNull(expected);
+
+            Assert.Equal(expected.AttributeId.OriginalString, actual.AttributeId.OriginalString);
+            Assert.Equal(expected.Category.OriginalString, actual.Category.OriginalString);
+            Assert.Equal(expected.Issuer, actual.Issuer);
+
+            AssertAttributeValueEqual((XacmlAttributeValue)expected.Property, (XacmlAttributeValue)actual.Property);
+        }
+
+        private static void AssertXAttributeEqual(XAttribute expected, XAttribute actual)
+        {
+            Assert.Equal(expected.Name, actual.Name);
+            Assert.Equal(expected.Value, actual.Value);
+        }
+
+        private static void AssertXElementEqual(XElement expected, XElement actual)
+        {
+            Assert.Equal(expected.Name, actual.Name);
+            Assert.Equal(expected.Value, actual.Value);
+        }
+
+        private static void AssertCollections<T>(ICollection<T> expected, ICollection<T> actual, Action<T, T> assertMethod)
```

Comment:
This looks a bit funny :) I guess you could use something like

```c#
Assert.Equal(expected.Count, expected.Intersect(actual, new AssertEqualityComparer<T>(assertMethod)).Count());
```
where `AssertEqualityComparer<T>` wraps `IEqualityComparer<T>` calling `assertMethod` for each element. But it probably will be less readable :)

Reply:
I tried using some of the existing AssertCollection implementations in unittest libs but after adding the dependencies it either ended up braking/colliding with existing libs, or would seem to require a very different approach to implementing comparer for the models. So ended up just making my own simple helper for iterating the collections.